src/main/java/org/kuali/kfs/module/cam/document/service/impl/GlLineServiceImpl.java

```diff
@@ -273,24 +273,44 @@ public CapitalAssetInformation findCapitalAssetInformation(String documentNumber
-                    groupAccountLine.getChartOfAccountsCode().equals(entry.getChartOfAccountsCode()) &&
-                    groupAccountLine.getAccountNumber().equals(entry.getAccountNumber()) &&
-                    groupAccountLine.getFinancialObjectCode().equals(entry.getFinancialObjectCode())) {
-            	// check that debit matches target acct lines and credit matches source accounting lines
-            	if((StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.SOURCE_ACCT_LINE_TYPE_CODE) && StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_CREDIT_CODE))
-            			|| (StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.TARGET_ACCT_LINE_TYPE_CODE) && StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_DEBIT_CODE))){
-            	matchingAssets.add(capitalAsset);
-            	  break;
-              }
+            if (groupAccountLine.getDocumentNumber().equals(entry.getDocumentNumber())
+                    && groupAccountLine.getChartOfAccountsCode().equals(entry.getChartOfAccountsCode())
+                    && groupAccountLine.getAccountNumber().equals(entry.getAccountNumber())
+                    && groupAccountLine.getFinancialObjectCode().equals(entry.getFinancialObjectCode())) {
+                // check that debit matches target acct lines and credit matches
+                // source accounting lines
+                boolean isGroupLineSource = StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.SOURCE_ACCT_LINE_TYPE_CODE);
+                boolean isGroupLineTarget = StringUtils.equals(groupAccountLine.getFinancialDocumentLineTypeCode(), KFSConstants.TARGET_ACCT_LINE_TYPE_CODE);
+                boolean isGLEntryCredit = StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_CREDIT_CODE);
+                boolean isGLEntryDebit = StringUtils.equals(entry.getTransactionDebitCreditCode(), KFSConstants.GL_DEBIT_CODE);
+                boolean isErrorCorrection = groupAccountLine.getAmount().isNegative();
+                logAddToCapitalAssetsMatchingLineTypeDetails(isGroupLineSource, isGroupLineTarget, isGLEntryCredit, isGLEntryDebit, isErrorCorrection);
+                if ((isGroupLineSource && isGLEntryCredit) || (isGroupLineTarget && isGLEntryDebit) || 
+                        (isErrorCorrection && isGroupLineSource && isGLEntryDebit) ||
+                        (isErrorCorrection && isGroupLineTarget && isGLEntryCredit)){
+                    matchingAssets.add(capitalAsset);
+                    break;
+                }
 
             }
         }
     }
+    
+    private void logAddToCapitalAssetsMatchingLineTypeDetails(boolean isGroupLineSource, boolean isGroupLineTarget, boolean isGLEntryCredit, 
+            boolean isGLEntryDebit, boolean isErrorCorrection) {
```

Comment:
This logging method doesn't seem to be adding quite enough value, since it prints the flags from the various checks but does not correlate them to the line that is being checked. I would suggest either adding some information about the line being examined, or just remove this method altogether.

Reply:
I'll see if I can add more details